Add a growing notification banner

forceunwrap opened the pull request:
Hey there.

I really like to use this library, but what I dislike is that one doesn't have the possibility to not use the scrolling labels when text is getting too long.
That's why I've implemented a new banner class which lets the view itself grow in size, instead of scrolling the labels in an animated fashion. I think I'm not the only one who could want this behavior, so I've created this pull request.

Here are some screenshots. You can also check it out, I've added it to the Example app.

<img width="526" alt="growing-portrait" src="https://user-images.githubusercontent.com/3381717/46574325-c06bad80-c9a1-11e8-8fb0-c66952fff09c.png">
<img width="1008" alt="growing-landscape" src="https://user-images.githubusercontent.com/3381717/46574326-c3669e00-c9a1-11e8-958c-9c20574fa895.png">

The labels are respecting the safe area, as the banner can grow all the way down to the notch, if there is much text about to be displayed.

There's still some work which has to be done before merge, i.e. considering the left/ right view for correct height calculation, if set. Before doing that I would like to know what you think about it.

Cheers

forceunwrap commented:
Hey @Daltron did you already have some time to check this out? What do you think? :)

Daltron commented:
Hi @forceunwrap! I'm really sorry for the delay! This feature looks great and I think a lot of people will appreciate having ready! Feel free to continue finishing it up and ping me when its ready for review and we will get this in the next release! Thanks for the great work! 👍

forceunwrap commented:
Hey @Daltron, nvm :)
The feature is now finished and ready for review. The only thing which was missing was support for left and right view which has now been added.

I've added two examples to the example app, one without and one with a side view.
Tested on iPad, various iPhone sizes (with and w/o notch) and iOS 10 - iOS 12.

Features of the GrowingNotificationBanner are:

- Title + Subtitle
- Left/ right view
- Determines it's height automatically based on the length of the labels, respecting left/ right views

For now I've dropped the support for attributed strings and custom views. The mechanism for determining the height requires the knowledge of the font, which is kinda hard to do with attributed strings I think. The custom view would have similar problems.

forceunwrap commented:
Hey @Daltron, have you had a chance to look into the finished feature?

Daltron commented:
@forceunwrap everything looks great! VERY, VERY nicely done! 👍

Daltron commented:
@forceunwrap Could you add some documentation to the README please and open up another PR?

forceunwrap commented:
@Daltron Yep, I already had that in mind. Will do it in the next few days. Thanks!

Daltron commented:
@forceunwrap Great, thank you! Also, CocoaPods 1.8.0 release now includes this feature. 👍
